dask_expr/io/parquet.py

```diff
@@ -71,6 +72,19 @@ def _tokenize_fileinfo(fileinfo):
     )
 
 
+_CPU_COUNT_SET = False
+
+
+def _maybe_adjust_cpu_count():
+    global _CPU_COUNT_SET
+    if not _CPU_COUNT_SET:
+        # Set the number of threads to the number of cores
+        # This is a default for pyarrow, but it's not set by default in
+        # dask/distributed
+        pa.set_cpu_count(os.cpu_count())
+        _CPU_COUNT_SET = True
```

Comment:
This is just the CPU thread pool. Arrow is distinguishing IO and CPU. The IO thread pool is set to smth like two times the number of logical CPUs and the IO pool does not respect the OMP_NUM_THREAD variable. We only have to do this because distributed is setting this to one.

Also your experiments ran on very different code